### PR TITLE
Animation pointer import support

### PR DESCRIPTION
**Pull request by pfcDorn**
**Added support for importing Animation Pointer:**
- Node Translate, Rotation and Scale _(pointer gets converted to using the non-animation-pointer code path)_
- BlendShapes _(pointer gets converted to using the non-animation-pointer code path)_
- Cameras
- Punctual Lights _(via new IImportAnimationPointerRootExtension Interface for Root Extensions)_
- Materials _(via new MaterialPropertiesRemapper* class assigned to AnimationPointerImportContext)_

**IImportAnimationPointerRootExtension Interface:**
- added to support Root Extensions for pointer data and value conversion
- currently only be used by the Punctual Lights Extension (KHR_LightsPunctualExtension)
- IExportAnimationPointerRootExtension should be added in the future to support custom Root Extensions for export

**MaterialPropertiesRemapper:**
- material property mapping and conversion class for Gltf Property <> Unity Property
- currently only be used by the Animation Pointer importer/exporter
- can also be used to extend it with custom mappings _(take a look in the "DefaultMaterialPropertiesRemapper"-Class and AnimationPointerImportContext/AnimationPointerExportContext)_

**Sample:**
[AnimateAllTheThings.zip](https://github.com/KhronosGroup/UnityGLTF/files/14852607/AnimateAllTheThings.zip)

### CODE REVIEW

File: Runtime/Scripts/SceneExporter/ExporterAnimationPointer.cs [hunk 1]
```diff
@@ -95,204 +76,40 @@ public void AddAnimationData(Object animatedObject, string propertyName, GLTFAni
...
 			switch (animatedObject)
 			{
 				case Material material:
 					// Debug.Log("material: " + material + ", propertyName: " + propertyName);
 					// mapping from known Unity property names to glTF property names
-					switch (propertyName)
+					if (animationPointerExportContext == null ||
+					    animationPointerExportContext.materialPropertiesRemapper == null)
 					{
-						case "_Color":
```
Comment (hybridherbst) [inline Runtime/Scripts/SceneExporter/ExporterAnimationPointer.cs:105]: Are these alternative property names still supported or did that support get lost in this PR?
As in: when animating e.g. an URP/Lit material, will that still export correctly with KHR_animation_pointer?

File: Editor/Scripts/GLTFImporterInspector.cs [hunk 2]
```diff
@@ -79,6 +79,7 @@ private void ModelInspectorGUI()
 			EditorGUILayout.LabelField("Scene", EditorStyles.boldLabel);
 			EditorGUILayout.PropertyField(serializedObject.FindProperty(nameof(GLTFImporter._removeEmptyRootObjects)));
 			EditorGUILayout.PropertyField(serializedObject.FindProperty(nameof(GLTFImporter._scaleFactor)));
+			EditorGUILayout.PropertyField(serializedObject.FindProperty(nameof(GLTFImporter._importCamera)));
```
Comment (hybridherbst) [inline Editor/Scripts/GLTFImporterInspector.cs:82]: I think this PR might be based on the camera import PR, is that intended? We might want to rebase once the camera PR has landed to make it more clear

File: Runtime/Plugins/GLTFSerialization/Schema/GLTFProperty.cs [hunk 3]
```diff
@@ -37,7 +37,8 @@ public static IReadOnlyList<string> RegisteredExtensions
 			{ KHR_draco_mesh_compression_Factory.EXTENSION_NAME, new KHR_draco_mesh_compression_Factory() },
 			{ KHR_texture_basisu_Factory.EXTENSION_NAME, new KHR_texture_basisu_Factory() },
 			{ EXT_meshopt_compression_Factory.EXTENSION_NAME, new EXT_meshopt_compression_Factory() },
-			{ EXT_mesh_gpu_instancing_Factory.EXTENSION_NAME, new EXT_mesh_gpu_instancing_Factory() }
+			{ EXT_mesh_gpu_instancing_Factory.EXTENSION_NAME, new EXT_mesh_gpu_instancing_Factory() },
+			{ KHR_animation_pointerExtensionFactory.EXTENSION_NAME, new KHR_animation_pointerExtensionFactory() }
 		};
```
Comment (hybridherbst) [inline Runtime/Plugins/GLTFSerialization/Schema/GLTFProperty.cs:42]: add comma at end of entry so future changes don't have to change the line again

File: Runtime/Plugins/GLTFSerialization/Utilities/GltfRootExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace GLTF.Utilities
+{
+    public static class GltfRootExtensions
+    {
+        public static int[] GetAllNodeIdsWithMaterialId(this GLTFRoot root, int id)
+        {
+            List<int> ids = new List<int>();
+
+            for (int iMeshes = 0; iMeshes < root.Meshes.Count; iMeshes++)
```
Comment (hybridherbst) [inline Runtime/Plugins/GLTFSerialization/Utilities/GltfRootExtensions.cs:13]: Can the Meshes array be null? Should we check for that here?
E.g. for a materials-only import file

File: Runtime/Plugins/GLTFSerialization/Utilities/GltfRootExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            }
+            
+            return ids.ToArray();
+        }
+
+        public static int[] GetAllNodeIdsWithCameraId(this GLTFRoot root, int id)
+        {
+            List<int> ids = new List<int>();
+
+            for (int i = 0; i < root.Nodes.Count; i++)
```
Comment (hybridherbst) [inline Runtime/Plugins/GLTFSerialization/Utilities/GltfRootExtensions.cs:41]: Can nodes be null? E.g. a materials-only file?

File: Runtime/Scripts/Plugins/AnimationPointer/MaterialPropertiesRemapper.cs [hunk 6]
```diff
@@ -0,0 +1,485 @@
+﻿using System;
+using System.Collections.Generic;
+using GLTF.Schema;
+using UnityEngine;
+
+namespace UnityGLTF.Plugins
+{
+    public class MaterialPointerPropertyMap
+    {
+        public enum PropertyTypeOption {LinearColor, SRGBColor, Texture, LinearTexture, TextureTransform, Float}
```
Comment (hybridherbst) [inline Runtime/Scripts/Plugins/AnimationPointer/MaterialPropertiesRemapper.cs:10]: I think we want that to be on multiple lines for easier future changes

File: Runtime/Scripts/Plugins/AnimationPointer/MaterialPropertiesRemapper.cs [hunk 7]
```diff
@@ -0,0 +1,485 @@
...
+                PropertyNames = new[] { "_EmissionMap_ST", "_EmissiveTexture_ST", "emissiveTexture_ST" },
+                GltfPropertyName =
+                    $"emissiveTexture/extensions/{ExtTextureTransformExtensionFactory.EXTENSION_NAME}/{ExtTextureTransformExtensionFactory.SCALE}",
+                GltfSecondaryPropertyName =
+                    $"emissiveTexture/extensions/{ExtTextureTransformExtensionFactory.EXTENSION_NAME}/{ExtTextureTransformExtensionFactory.OFFSET}",
+                ExtensionName = ExtTextureTransformExtensionFactory.EXTENSION_NAME
+            };
+            AddMap(emissiveTexture);
+
+            /*var roughnessTex = new MaterialPointerPropertyMap
```
Comment (hybridherbst) [inline Runtime/Scripts/Plugins/AnimationPointer/MaterialPropertiesRemapper.cs:324]: Is this intentionally missing?

File: Runtime/Scripts/SceneExporter/ExporterAnimationPointer.cs (hunk 1 above)
Comment (hybridherbst) [reply Runtime/Scripts/SceneExporter/ExporterAnimationPointer.cs:105]: Ah, I see it in DefaultMaterialPropertiesRemapper

File: Runtime/Scripts/Plugins/AnimationPointer/MaterialPropertiesRemapper.cs (hunk 7 above)
Comment (pfcDorn) [reply Runtime/Scripts/Plugins/AnimationPointer/MaterialPropertiesRemapper.cs]: Updated it in: 15542c7e0755edec44ffb5f1c42f6aba13c57fce

File: Runtime/Scripts/Plugins/AnimationPointer/MaterialPropertiesRemapper.cs (hunk 6 above)
Comment (pfcDorn) [reply Runtime/Scripts/Plugins/AnimationPointer/MaterialPropertiesRemapper.cs]: Changed with 9de922c583e869c97aa2417986cb7240dca19752

File: Runtime/Plugins/GLTFSerialization/Utilities/GltfRootExtensions.cs (hunk 5 above)
Comment (pfcDorn) [reply Runtime/Plugins/GLTFSerialization/Utilities/GltfRootExtensions.cs:49]: Changed with 427a33216374fd9a6fd7ce222ff7a9e0293f7463

File: Runtime/Plugins/GLTFSerialization/Utilities/GltfRootExtensions.cs (hunk 4 above)
Comment (pfcDorn) [reply Runtime/Plugins/GLTFSerialization/Utilities/GltfRootExtensions.cs:16]: Changed with 427a33216374fd9a6fd7ce222ff7a9e0293f7463

File: Runtime/Plugins/GLTFSerialization/Schema/GLTFProperty.cs (hunk 3 above)
Comment (pfcDorn) [reply Runtime/Plugins/GLTFSerialization/Schema/GLTFProperty.cs:42]: will be changed in the Dispersion branch with commit 5de6b1d06efda04df7635b3440a7f9309e291529

**Pull request merged**